hugegraph-core/src/main/java/org/apache/hugegraph/traversal/optimize/TraversalUtil.java

```diff
@@ -105,12 +107,71 @@ public static HugeGraph getGraph(Step<?, ?> step) {
     }
 
     public static HugeGraph tryGetGraph(Step<?, ?> step) {
-        Graph graph = step.getTraversal().getGraph().get();
-        if (graph instanceof HugeGraph) {
-            return (HugeGraph) graph;
+        Optional<Graph> graph = step.getTraversal()
+                                    .getGraph()
+                                    .filter(g -> {
+                                        return !(g instanceof EmptyGraph);
```

Comment:
add "TODO: remove these EmptyGraph judgments when upgrading tinkerpop to a [fixed version](https://github.com/apache/tinkerpop/pull/1699)"